packages/tools/playground/src/tools/monacoManager.ts

```diff
@@ -220,126 +210,33 @@ class Playground {
-            code += this._getCode("useBox");
-            if (this._templates.length && useShadows && useLight != "Hemispheric") {
-                code += "shadowGenerator.addShadowCaster(box);\n";
-            }
-        }
-        if (useCylinder) {
-            code += this._getCode("useCylinder");
-            if (this._templates.length && useShadows && useLight != "Hemispheric") {
-                code += "shadowGenerator.addShadowCaster(cylinder);\n";
-            }
-        }
-
-        // Animations
-        if (useAnimation != "None") {
-            code += "\n// Import animated character :\n";
-            code += this._getCode(useAnimation);
-            if (this._templates.length && useShadows && useLight != "Hemispheric") {
-                code = code.slice(0, code.length - 4) + "    shadowGenerator.addShadowCaster(hero);\n});\n";
+    private _insertCodeAtCursor(code: string) {
+        if (this._editor) {
+            // Get the current position of the cursor
+            const position = this._editor?.getPosition();
+            if (position) {
+                // Fix indent regarding current position
+                if (position.column && position.column > 1) {
+                    code = this._indentCode(code, position.column - 1).slice(position.column - 1);
+                }
+                // Insert code
+                this._editor?.executeEdits("", [
+                    {
+                        range: new monaco.Range(position.lineNumber, position.column, position.lineNumber, position.column),
+                        text: code,
+                        forceMoveMarkers: true,
+                    },
+                ]);
             }
         }
+    }
 
-        // Particles
-        if (useParticles != "None") {
-            code += "\n// Create particle system :\n";
-            code += this._getCode(useParticles);
-        }
-
-        // Snippets
-        if (useSnippet != "None") {
-            code += "\n// Load snippet from server :\n";
-            code += this._getCode(useSnippet);
-        }
-
-        // Import / Export
-        if (useImport != "None") {
-            code += "\n// Import or Export meshes :\n";
-            code += this._getCode(useImport);
-        }
-
-        code += "\n// Return\nreturn scene;";
-
-        // Encapsulate function (Javascript)
-        let all_code = "";
-        if (this.globalState.language === "JS") {
-            all_code = "var createScene = function () {" + this._indentCode(code, 4) + "\n};";
-
-            // Encapsulate function (Typescript)
-        } else {
-            all_code += "class Playground {\n";
-            all_code += "    public static CreateScene(engine: BABYLON.Engine, canvas: HTMLCanvasElement): BABYLON.Scene {" + this._indentCode(code, 8) + "\n";
-            all_code += "    }\n";
-            all_code += "}";
+    private _insertSnippet() {
+        const snippetKey = Utilities.ReadStringFromStore("snippetKey", "");

```

Comment:
why are you using the localstorage to get the snippet key? couldn't this be done using the global scope?

Reply:
Good catch, I admit that I am not 100% happy with this.
Here is why :
___
Short answer : I did it to stay as close as possible to previous structure which is 100% based on localStorage, except for direct `onClick()` triggers which are defined without any parameter. Basically the goal was to reduce as much as possible number of files/lines to be modified to have it working
___
More details :
So basically this `React` UI seems to be built in a way where either you have a button and it trigger a single function (that I could have prepared to be called with right param) either you send a list of subitems and here you don't have the hand on the subItem to be "onClicked"

To avoid using localStorage, we would have to :
- Modify `commandDropdownComponent.tsx`, the loop `m.subItems.map((s, index)` line 129 would need to send the item/index to the `onClick` function, I am not sure of the possible result on others elements of the PG UI which are not considered by this feature
- Modify as well `commandBarComponent.tsx`, to take the element in the `onClick` function
- Modify all the pile of calls which goes from this `onClick`, i.e. adding params, in the `onInsertSnippet`, and as well the `onInsertSnippetRequiredObservable.notifyObservers()` etc.
___
Also since I was not sure this code would be keeped (as, already, I'm deleting a lot of lines from the previous commit 2 days ago...) I did not go through this, even if I agree would be neater :)

Reply:
@RaananW tell me if you prefer that I review and have the snippetKey passed by another way :)

Reply:
The snippet key doesn't need to persist, if I understand correctly. So storing it in the globalstate will be more than enough. It will reset with a restart, but until then, and while working it will persist. Does it work?

Reply:
Absolutely. the snippetKey doesn't need to persist since it's always used just after being set. Just going though globalState should work.